advisor/cmd/gen.go

```diff
@@ -11,18 +11,50 @@ var genCmd = &cobra.Command{
 	Short: "Generate resources",
 }
 
+var (
+	allNamespaces  bool
+	allInNamespace bool
+)
+
+func init() {
+	networkPolicyCmd.Flags().BoolVarP(&allNamespaces, "all-namespaces", "A", false, "Generate policies for all pods in all namespaces")
+	networkPolicyCmd.Flags().BoolVar(&allInNamespace, "all", false, "Generate policies for all pods in the current namespace")
+}
+
 var networkPolicyCmd = &cobra.Command{
 	Use:     "networkpolicy [pod-name]",
 	Aliases: []string{"netpol"},
 	Short:   "Generate network policy",
-	Args:    cobra.ExactArgs(1),
+	Args:    cobra.MaximumNArgs(1),
```

Comment:
Changed this because when using `--all` or `-A` you won't need to pass a <pod-name> arg so it should be `<= 1` Args are possibly valid, then just check if it's not set.